pkgs/servers/web-apps/lemmy/update.sh

```diff
@@ -14,7 +14,8 @@ const latest_rev = $(curl -q https://api.github.com/repos/${owner}/${server_repo
   jq -r '.tag_name')
 const latest_version = $(echo $latest_rev)
 const current_version = $(jq -r '.version' $directory/pin.json)
-if (latest_version === $current_version) {
+echo "latest version: $latest_version, current version: $current_version"
+if ($latest_version == $current_version) {
```

Comment:
```suggestion
if (latest_version === current_version) {
```
The update script doesn't work for me unless it's changed to this.

Looks good to me other than this

Reply:
Interesting. For me that fails completely :D 

```
. builtin echo 'latest version: 0.17.3, current version: 0.17.3'
latest version: 0.17.3, current version: 0.17.3
  if (latest_version === current_version) {

./pkgs/servers/web-apps/lemmy/update.sh:18: Syntax error in expression (near Id.Arith_Equal)
```
It works the way it's in this branch tho.

Reply:
I've made a couple of tests, it looks like a difference introduced in one of the latest versions of oil.
You guys might be on two different versions of oil.
I realise now that making a script with a non standard shell can be confusing.
What Bill is suggesting is the recommended way with oil 0.15.0 it seems.
you can both try `nix-shell  -p oil` and `oil --version` to check the versions.
if you just run `oil`, you'll get an interactive shell.
you can both verify that
```
const latest_version = "0.17.3"
const current_version = "0.17.2"
if ("$latest_version" === "$current_version") { echo "true" } else { echo "false" }
```
I'm guessing it's a more future proof way of writing this
let me know if this fails.

Reply:
```
  if ("$latest_version" === "$current_version") {
                          ^
./pkgs/servers/web-apps/lemmy/update.sh:18: Syntax error in expression (near Id.Arith_Equal)
``` 
I'm not sure why my nixos machine is on oil 0.9 tho, since it's tracking unstable 😕 on nix-darwin I get '0.12.4' for `nix-shell -p oil`

but that version still fails:

```
❯ oil                                                                                                                                                                                      
$ oil
    if ("$latest_version" === "$current_version") {
>                           ^
> ./pkgs/servers/web-apps/lemmy/update.sh:18: Syntax error in expression (near Id.Arith_Equal)
  ./pkgs/servers/web-apps/lemmy/update.sh:18: Syntax error in expression (near Id.Arith_Equal)
                                                                               ^~
[ interactive ]:3: Syntax error in expression (near Id.Expr_Name)
```

Reply:
I have version `Oil version 0.12.9`, the code above works for me.
I'd be fine changing it to this:

```
❯ nix-shell -p oil --run oil
oil$ const latest_version = "0.17.3"
oil$ const current_version = "0.17.2"
oil$ if ("$latest_version" === "$current_version") { echo "true" } else { echo "false" }
false
oil$ 
```


>  I realise now that making a script with a non standard shell can be confusing.

Ah but that's also where all the fun is. We knew we'd run into something like this eventually.

Reply:
Oil on unstable is at version 0.15.0 lol
Lets change it to this then
if ("$latest_version" === "$current_version") 

If by the next version update this still fails for you @marcusramberg lets brainstorm again what we can do about it.